### PR TITLE
Use `@iobroker/adapter-dev` for development-related tasks

### PR DESCRIPTION
**Pull request by UncleSamSwiss**
**PR Checklist:**

-   [x] Provide a meaningful description to this PR or mention which issues this fixes.
-   [x] Add tests for your change. This includes negative tests (i.e. inputs that need to fail) as well as baseline tests (i.e. how should the directory structure look like?).
-   [x] Run the test suite with `npm test`
-   [x] If there are baseline changes, review them and make a separate commit for them with the comment "accept baselines" if they are desired changes
-   [x] Ensure the project builds with `npm run build`
-   [ ] If you added a required option, also add it to the template creation (`.github/create_templates.ts`)
-   [x] Add a detailed migration description to `docs/updates` explaining what the user needs to do when manually updating an existing project
-   [x] Add your changes to `CHANGELOG.md` (referencing the migration description and this PR or the issue you fixed)

**Description:**  
- Using `@iobroker/adapter-dev` instead of `gulp` for translations in generated adapters
- Using `@iobroker/adapter-dev` internally in create-adapter when doing translations
- Offer the possibility to create the `i18n/<lang>/translations.json` files directly from create-adapter

### CODE REVIEW

File: docs/updates/20211018_adapter-dev.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "translate": "translate-adapter",
+```
+
+## New commands
+
+Now you can use the `translate` script instead of `gulp`:
+
+| Old (gulp)                       | New (adapter-dev)             |
+| -------------------------------- | ----------------------------- |
+| `gulp translate`                 | `npm run translate translate` |
```
Comment (AlCalzone) [inline docs/updates/20211018_adapter-dev.md:27]: See https://github.com/ioBroker/adapter-dev/issues/58 :)

File: docs/updates/20211018_adapter-dev.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+## Migration
+
+Simply follow these steps to switch from `gulp` to `@iobroker/adapter-dev`:
+
+1. Remove the `gulpfile.js` in the root directory
+1. Remove the following dev-dependencies from your `package.json`:
+    - `gulp`
+    - `@types/gulp` (if available)
+    - `axios` (except if you are using it in other development tasks)
```
Comment (AlCalzone) [inline docs/updates/20211018_adapter-dev.md]:
```suggestion
    - `axios` (except if you are using it in other development tasks or in production)
```
I know you wrote devDependencies above, but better be sure :)

File: src/lib/core/questions.ts [hunk 3]
```diff
@@ -642,6 +642,25 @@ export const questionGroups: QuestionGroup[] = [
...
+				],
+				type: "select",
+				name: "i18n",
+				label: "Translation Management",
+				optional: true,
+				expert: true,
+				message:
+					"Would you like to use JSON files (required for Weblate) or directly words.js to manage translations?",
+				initial: "words.js",
+				choices: ["words.js", "JSON"],
```
Comment (AlCalzone) [inline src/lib/core/questions.ts]: I would reformulate:
"How would you like to manage translations?"
* JSON files (hint = recommended, required for Weblate)
* words.js (hint = legacy)

File: templates/admin/src/i18n/zh-cn.json.ts [hunk 4]
```diff
@@ -1,3 +1,3 @@
 import { getI18nJsonTemplate } from "../../../../src/lib/translation";
 
-export = getI18nJsonTemplate("zh-cn");
+export = getI18nJsonTemplate("zh-cn", true);
```
Comment (AlCalzone) [inline templates/admin/src/i18n/zh-cn.json.ts:3]: Expanding a single template into an array would be handy now :D

**Review by AlCalzone — CHANGES_REQUESTED** [review]
Just some minor things

File: docs/updates/20211018_adapter-dev.md (hunk 1 above)
Comment (UncleSamSwiss) [reply docs/updates/20211018_adapter-dev.md:27]: Agreed, will go into a future release of adapter-dev.

File: templates/admin/src/i18n/zh-cn.json.ts (hunk 4 above)
Comment (UncleSamSwiss) [reply templates/admin/src/i18n/zh-cn.json.ts:3]: Yet again, yes.

File: src/lib/core/questions.ts (hunk 3 above)
Comment (UncleSamSwiss) [reply src/lib/core/questions.ts]: But we still keep words.js as the default, right?

Comment (AlCalzone) [reply src/lib/core/questions.ts]: I have no preference there.

Comment (AlCalzone) [reply src/lib/core/questions.ts]: Actually, I'm a bit confused now. I thought `words.js` is the only way for non-react and JSON files the recommended way for React and mandatory for Weblate. Should we even offer this question then?

Comment (UncleSamSwiss) [reply src/lib/core/questions.ts]: For regular HTML UIs you need to have a `words.js`, but you can also (additionally) manage the translations in JSON files. This is required for Weblate.
For React UIs you can only use JSON - for this reason the question is not asked if the user chooses to use a React UI (see line 648).

Comment (AlCalzone) [reply src/lib/core/questions.ts]: Ok the question now sounds like these are exclusive options, which isn't optimal. How about:
"How would you like to manage translations?"
* _Only words.js_ (default)
* _words.js and JSON files_ (hint = required for Weblate)

Comment (UncleSamSwiss) [reply src/lib/core/questions.ts]: I know we are splitting hairs here, but I agree with you that the question must be clearer. For me "managing" is where you write your translations. The fact that words.js is still used with JSON files shouldn't be so prominent. How about keeping the texts as before and just add something about `words.js` to the hint of `JSON`?

Comment (AlCalzone) [reply src/lib/core/questions.ts]: That could work too.

**Comment by UncleSamSwiss** [comment]
@AlCalzone Clarified question about words.js once again. I think now this PR is ready to merge.

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**